cupy/fft/fft.py

```diff
@@ -175,6 +180,45 @@ def _fft(a, s, axes, norm, direction, value_type='C2C', overwrite_x=False,
     return a
 
 
+@cupy.util.memoize()
```

Comment:
I agree that not allowing axes as a list is problematic.

I will just remove `memoize` from this function.

Reply:
Hi @toslunar @grlee77 I was tracing back to the commit a1d7e60 and found this discussion. 

I think the concern 1 (unbounded combinations) is probably too much, because other CuPy functions that already use `@memoize()` can also have many many possible combinations, and yet we're using it without noticeable performance issue. The key here is the actual combinations that one can run into in each runtime session is likely small. 

For concern 2, what if we make `axes` a tuple before calling this function? It's an invariant in this function anyway.